apps/extension/client/src/initiateEditor.ts

```diff
@@ -129,25 +126,57 @@ export const initiateEditor = (
     }
   };
 
-  context.subscriptions.push(
-    vscode.workspace.onDidSaveTextDocument(async (document) => {
-      const text = document.getText();
-      const result = parseMachinesFromFile(text);
-
-      if (result.machines.length > 0) {
-        result.machines.forEach((machine, index) => {
-          sendMessage({
-            type: "RECEIVE_CONFIG_UPDATE_FROM_VSCODE",
-            config: machine.toConfig({ hashInlineImplementations: true })!,
-            index: index,
-            uri: resolveUriToFilePrefix(document.uri.path),
-            layoutString: machine.getLayoutComment()?.value || "",
-            implementations: getInlineImplementations(machine, text),
-          });
+  const sendChangesToVisualEditor = (document: vscode.TextDocument) => {
+    const text = document.getText();
+    const result = parseMachinesFromFile(text);
+
+    if (result.machines.length > 0) {
+      result.machines.forEach((machine, index) => {
+        sendMessage({
+          type: "RECEIVE_CONFIG_UPDATE_FROM_VSCODE",
+          config: machine.toConfig({ hashInlineImplementations: true })!,
+          index: index,
+          uri: resolveUriToFilePrefix(document.uri.path),
+          layoutString: machine.getLayoutComment()?.value || "",
+          implementations: getInlineImplementations(machine, text),
         });
-      }
-    })
-  );
+      });
+    }
+  };
+
+  const getSendChangesSubscription = () => {
+    const xstateConfig = vscode.workspace.getConfiguration("xstate");
+    const sendChangesOnCreation =
+      xstateConfig.get<string>("sendChanges") === "onCreation";
+    const sendChangesDelay =
+      xstateConfig.get<number>("sendChangesDelay") ?? 1000;
+
+    if (sendChangesOnCreation) {
+      const debouncedSendChangesToVisualEditor = debounce(
+        sendChangesToVisualEditor,
+        sendChangesDelay
+      );
+
+      return vscode.workspace.onDidChangeTextDocument(({ document }) =>
+        debouncedSendChangesToVisualEditor(document)
+      );
+    } else {
+      return vscode.workspace.onDidSaveTextDocument(sendChangesToVisualEditor);
```

Comment:
Perhaps this should still result in an immediate refresh of the editor even if the refreshing strategy is set to typing?